felix/bpf/bpfmap/bpf_maps.go

```diff
@@ -34,101 +32,99 @@ import (
 	"github.com/projectcalico/calico/felix/bpf/state"
 )
 
-func CreateBPFMapContext(
-	ipsetsMapSize,
-	natFEMapSize,
-	natBEMapSize,
-	natAffMapSize,
-	routeMapSize,
-	ctMapSize,
-	ifstateSize int,
-	repinEnabled bool,
-) *bpf.MapContext {
-	bpfMapContext := &bpf.MapContext{
-		RepinningEnabled: repinEnabled,
-		MapSizes:         map[string]uint32{},
-	}
-	bpfMapContext.MapSizes[ipsets.MapParameters.VersionedName()] = uint32(ipsetsMapSize)
-	bpfMapContext.MapSizes[nat.FrontendMapParameters.VersionedName()] = uint32(natFEMapSize)
-	bpfMapContext.MapSizes[nat.BackendMapParameters.VersionedName()] = uint32(natBEMapSize)
-	bpfMapContext.MapSizes[nat.AffinityMapParameters.VersionedName()] = uint32(natAffMapSize)
-	bpfMapContext.MapSizes[routes.MapParameters.VersionedName()] = uint32(routeMapSize)
-	bpfMapContext.MapSizes[conntrack.MapParams.VersionedName()] = uint32(ctMapSize)
-
-	bpfMapContext.MapSizes[state.MapParameters.VersionedName()] = uint32(state.MapParameters.MaxEntries)
-	bpfMapContext.MapSizes[arp.MapParams.VersionedName()] = uint32(arp.MapParams.MaxEntries)
-	bpfMapContext.MapSizes[failsafes.MapParams.VersionedName()] = uint32(failsafes.MapParams.MaxEntries)
-	bpfMapContext.MapSizes[nat.SendRecvMsgMapParameters.VersionedName()] = uint32(nat.SendRecvMsgMapParameters.MaxEntries)
-	bpfMapContext.MapSizes[nat.CTNATsMapParameters.VersionedName()] = uint32(nat.CTNATsMapParameters.MaxEntries)
-	bpfMapContext.MapSizes[ifstate.MapParams.VersionedName()] = uint32(ifstateSize)
-
-	return bpfMapContext
+type Maps struct {
+	IpsetsMap       bpf.Map
+	StateMap        bpf.Map
+	ArpMap          bpf.Map
+	FailsafesMap    bpf.Map
+	FrontendMap     bpf.Map
+	BackendMap      bpf.Map
+	AffinityMap     bpf.Map
+	RouteMap        bpf.Map
+	CtMap           bpf.Map
+	SrMsgMap        bpf.Map
+	CtNatsMap       bpf.Map
+	IfStateMap      bpf.Map
+	RuleCountersMap bpf.Map
 }
 
-func MigrateDataFromOldMap(mc *bpf.MapContext) {
-	ctMap := mc.CtMap
-	err := ctMap.CopyDeltaFromOldMap()
-	if err != nil {
-		log.WithError(err).Debugf("Failed to copy data from old conntrack map %s", err)
+func (m *Maps) Destroy() {
+	maps := []bpf.Map{
+		m.IpsetsMap,
+		m.StateMap,
+		m.ArpMap,
+		m.FailsafesMap,
+		m.FrontendMap,
+		m.BackendMap,
+		m.AffinityMap,
+		m.RouteMap,
+		m.CtMap,
+		m.SrMsgMap,
+		m.CtNatsMap,
 	}
```

Comment:
QQ: for symmetry did we want/need to include `IfStateMap` and `RuleCountersMap` ?

Reply:
not now, i want to keep it the way it used to be, it is unrelated.